### ISSUE TITLE
Incorrect warning about Emojis not wrapped in <span> when using styled components

### ISSUE THREAD

**Issue by tijsg** [jsx-eslint/eslint-plugin-jsx-a11y#736]
**Observed behaviour:**

When wrapping an Emoji in a \<span> styled component with the styled-components npm package, e.g.: 

\<StyledEmoji role="img" aria-label="information">
            ℹ️
\</StyledEmoji>

const StyledEmoji = styled.span\`
  padding-right: 10px;
\`;

It incorrectly triggers a warning that it's not wrapped in \<span> with the following message:
warning  Emojis should be wrapped in <span>, have role="img", and have an accessible description    
with aria-label or aria-labelledby  jsx-a11y/accessible-emoji

**Expected behaviour:**

No warning should be displayed in this scenario

**Comment by ljharb** [comment]
Your StyledSpan there also doesn't have role img or an aria-label, though.

**Comment by tijsg** [comment]
> Your StyledSpan there also doesn't have role img or an aria-label, though.

updated description with the actual code that produces it

**Comment by ljharb** [comment]
There's not really a practical way for a linter to know for sure that `StyledSpan` results in a `span`; we have a precedent to maybe allow you to configure things so that "StyledSpan"s are treated as "span"s, but i'm not sure if that's hugely valuable here, since you wouldn't actually want multiple spans across your app to all have a right padding (and you definitely wouldn't want all emojis to have that, especially when they're inline with other text)
